sklearn/linear_model/tests/test_coordinate_descent.py

```diff
@@ -628,3 +629,26 @@ def test_sparse_dense_descent_paths():
         _, coefs, _ = path(X, y, fit_intercept=False)
         _, sparse_coefs, _ = path(csr, y, fit_intercept=False)
         assert_array_almost_equal(coefs, sparse_coefs)
+
+
+def test_check_input_false():
+    X, y, _, _ = build_dataset(n_samples=20, n_features=10)
+    X = check_array(X, order='F', dtype='float64')
+    y = check_array(X, order='F', dtype='float64')
+    clf = ElasticNet(selection='cyclic', tol=1e-8)
+    # Check that no error is raised if data is provided in the right format
+    clf.fit(X, y, check_input=False)
+    X = check_array(X, order='F', dtype='float32')
```

Comment:
I guess that if you try with the correct dtype but the wrong memory layout:

```python
X = check_array(X, order='C', dtype='float64')
```

you should not get an error but a wrong results. This is expected but maybe this should be asserted in the tests as well.